src/operator/distinctUntilChanged.ts

```diff
@@ -10,13 +11,15 @@ import {errorObject} from '../util/errorObject';
  * @param {function} [compare] optional comparison function called to test if an item is distinct from the previous item in the source.
  * @returns {Observable} an Observable that emits items from the source Observable with distinct values.
  */
-export function distinctUntilChanged<T>(compare?: (x: any, y: any) => boolean, keySelector?: (x: T) => any) {
-  return this.lift(new DistinctUntilChangedOperator(compare, keySelector));
+export function distinctUntilChanged<T>(compare?: (x: T, y: T) => boolean): Observable<T>;
+export function distinctUntilChanged<T, K>(compare: (x: K, y: K) => boolean, keySelector?: (x: T) => K): Observable<T>;
+export function distinctUntilChanged<T, K>(compare: (x: K, y: K) => boolean, keySelector?: (x: T) => K): Observable<T> {
```

Comment:
If you supply a `keySelector`, then the values that get handed to compare, are the type that is returned by the `keySelector` otherwise the values handed to compare are `T`.